oximeter/instruments/src/kstat/sampler.rs

```diff
@@ -1094,14 +1135,18 @@ impl KstatSamplerWorker {
 /// A type for reporting kernel statistics as oximeter samples.
 #[derive(Clone, Debug)]
 pub struct KstatSampler {
+    log: Logger,
     samples: Arc<Mutex<BTreeMap<TargetId, Vec<Sample>>>>,
     outbox: mpsc::Sender<Request>,
-    self_stat_rx: Arc<Mutex<mpsc::Receiver<Sample>>>,
+    self_stat_rx: Arc<Mutex<broadcast::Receiver<Sample>>>,
```

Comment:
It might be worth including a comment here explaining why we use an `Arc<Mutex<broadcast::Receiver<...>>>` here, rather than cloning this type by [resubscribing](https://docs.rs/tokio/latest/tokio/sync/broadcast/struct.Receiver.html#method.resubscribe) to the broadcast channel (as `broadcast` is a multi-producer, multi-consumer channel).

For the record, I think that the current code is definitely correct, and resubscribing would be wrong: each new `Receiver` has its own position in the channel, and we would like to have a single read position and are instead using the `broadcast` channel for ring-buffer-y bounded channel behavior. But, it might be worth writing that down for future readers?

Reply:
broadly, i think a comment explaining the use of `broadcast` rather than a `mpsc` channel here could be worthwhile, since we're not actually using it for its' MPMC aspect...

Reply:
Added a note in f303f1022